test/browser/home-page.test.jsx

```diff
@@ -4,9 +4,107 @@ var React = require('react/addons');
 var TestUtils = React.addons.TestUtils;
 
 var stubContext = require('./stub-context.jsx');
+var StubRouter = require('./stub-router');
 var HomePage = require('../../pages/home.jsx');
 
 var stubBlogFeedLoader = require('./stub-blog-feed-loader');
+var Util = require('../util.js');
+
+describe("HomePage", function() {
+ var homePage, pledgeBtn;
+
+ beforeEach(function() {
+  homePage =  stubContext.render(HomePage);
+  pledgeBtn = TestUtils.scryRenderedDOMComponentsWithClass(homePage, "icon-button")[0];
+ });
+
+ afterEach(function() {
+  stubContext.unmount(homePage);
+ });
+
+ it("does not show pledge modal by default", function() {
+  TestUtils.scryRenderedDOMComponentsWithClass(homePage, "modal-pledge").length.should.equal(0);
+ });
+
+ it("shows pledge modal when 'Pledge to Teach' button is clicked", function() {
+  TestUtils.Simulate.click(TestUtils.findRenderedDOMComponentWithTag(pledgeBtn, "a"));
+  homePage.context.showModal.callCount.should.equal(1);
```

Comment:
I tried using `TestUtils.scryRenderedDOMComponentsWithClass(homePage, "modal").length.should.equal(1);` at first but it failed 

@toolness could you help me understand why?